test/test_common.h

```diff
@@ -7,4 +7,126 @@
 #ifndef __RPMI_TEST_COMMON__
 #define __RPMI_TEST_COMMON__
 
+#include <stdio.h>
+#include <librpmi.h>
+#include <stdlib.h>
+
+/*
+ * Qemu shared memory offsets.
+ * its better to decode these data from DTS file, hardcoding for now.
+ */
+
+#define RPMI_SHM_XPORTS       (2)
+/*
+ * test assums qemu RPMI shared memory address is mapped at the following
+ * offset in gues RAM
+ */
+#define RPMI_SHM_OFSET       (0x0000000010001000)
+
+/* Total Shared memory size */
+#define SHM_SZ (512 * 1024 * 1024)
+
+#define RPMI_SHM_SZ (8 * 1024)
+
+/* Fast channel memory offset */
+#define FCM_OFFSET (0 * 256 * 1024)
+
+/* Sideband channel memory offset */
+#define SBM_OFFSET (0 * 256 * 1024)
+
+#define RPMI_SLOT_SIZE          64
+#define RPMI_MSG_MAX_DATA_SIZE (RPMI_SLOT_SIZE -  RPMI_MSG_HDR_SIZE)
+
+#define RPMI_MIN_QUEUE_SIZE             (16 * RPMI_SLOT_SIZE)
+#define RPMI_MIN_TOTAL_QUEUES_SIZE      (RPMI_QUEUE_MAX * \
+					 RPMI_MIN_QUEUE_SIZE)
+#define RPMI_MAX_SERVICE_GROUPS 16
+
+#define NUM_XPORTS_PER_SOC	(1)
+#define NUM_XPORTS(num_socks)	(num_socks + NUM_XPORTS_PER_SOC)
+
+struct group_config {
+	struct {
+		rpmi_uint32_t num_harts;
+		rpmi_uint32_t num_sockets;
+	} machine;
+
+	struct {
+		rpmi_uint32_t num_xports;
+		rpmi_uint32_t max_num_groups;
+	} rpmi;
+
+	struct {
+		rpmi_uint16_t vendor_id;
+		rpmi_uint16_t vendor_sub_id;
+		rpmi_uint32_t hw_info_len;
+		const rpmi_uint8_t *hw_info;
+	} base;
+	char *priv;
+	int priv_len;
+};
+
+struct rpmi_test {
+	char name[64];
+	void *rpmi_xport;
+	int arg0;
+	int resp0;
+	struct {
+		unsigned int svc_grp_id;
+		unsigned int svc_id;
+		unsigned int svc_type;
+	} attrs;
+	int query_data_len;
+	unsigned char query_data[RPMI_MSG_MAX_DATA_SIZE];
+	int resp_data_len;
+	unsigned char resp_data[RPMI_MSG_MAX_DATA_SIZE];
+
+	/* test specific function handlers*/
+	int (*init)(struct rpmi_test *test);
```

Comment:
For all the test callbacks, just pass rpmi_test_group pointer which already has rpmi_context and rpmi_transport pointers.

Reply:
No, rpmi_test still needs as we still need to track, request/response data per service group tests.